lib/Dialect/Torch/IR/TorchOps.cpp

```diff
@@ -6033,6 +6033,35 @@ LogicalResult AtenRot90Op::verify() {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// AtenCountNonzeroDimIntListOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult AtenCountNonzeroDimIntListOp::verify() {
+
+  auto selfType = cast<BaseTensorType>(getSelf().getType());
+
+  if (!selfType.hasDtype() || !selfType.hasSizes())
+    return success();
+
+  SmallVector<int64_t> dims;
+  if (!matchPattern(getDim(), m_TorchListOfConstantInts(dims)))
+    return emitOpError("expected dim to be constructed from list construct");
+  if (!isa<Torch::IntType>(getDim().getType().getContainedType()))
+    return emitOpError("list elements must be int type");
```

Comment:
This separate check is not required. Since, the above patern matching covers that.